### PR TITLE
Update models to actually reflect real relationships

### PR DESCRIPTION
**Pull request by jlhogan**
makes required Ids required, and non-nullable ids non-nullable by
removing cascade delete behavior, replacing it with restrict. Lots of
changes, and means we have to supply default values for ids on object
creation but i think we abstract all that into utils and this is better

### CODE REVIEW

File: src/Hedwig/ClientApp/src/generated/models/CdcReport.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export interface CdcReport {
      * @type {number}
      * @memberof CdcReport
      */
-    organizationId?: number;
+    organizationId: number;
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/generated/models/CdcReport.ts]: Do you know why this has no `?` but in child there is one?

File: src/Hedwig/Models/Family.cs [hunk 2]
```diff
@@ -16,9 +19,7 @@ public class Family : TemporalEntity, IValidateable
 		public bool Homelessness { get; set; }
 		public ICollection<FamilyDetermination> Determinations { get; set; }
 
-		// Optional FK to prevent cascade delete
-		// (multiple cascade delete FKs disallowed by SQLServer due to potential for cycles)
-		public int? OrganizationId { get; set; }
+		public int OrganizationId { get; set; }
```
Comment (cailyncodes) [inline src/Hedwig/Models/Family.cs]: Should this be required?

File: src/Hedwig/ClientApp/src/generated/models/CdcReport.ts (hunk 1 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/generated/models/CdcReport.ts]: right now, assumption in creating child/family is that user does _not_ provide the orgid, and we pull it from path params; i think we want to do this for report as well, so will mark orgId _not_ required on organizationReport

File: src/Hedwig/Models/Family.cs (hunk 2 above)
Comment (jlhogan) [reply src/Hedwig/Models/Family.cs]: see above comment. either/or

File: src/Hedwig/ClientApp/src/generated/models/Child.ts [hunk 3]
```diff
@@ -46,24 +46,12 @@ import {
...
-     * @memberof Child
-     */
-    organization?: Organization;
     /**
      * 
      * @type {string}
      * @memberof Child
      */
-    id?: string;
+    id: string;
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/generated/models/Child.ts]: We have three different types here and I'm not sure I know the difference.

Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/generated/models/Child.ts]: id, sasid, firstName that is

File: src/Hedwig/Models/FundingSpace.cs [hunk 4]
```diff
@@ -1,14 +1,17 @@
 using Newtonsoft.Json;
 using Newtonsoft.Json.Converters;
+using System.ComponentModel.DataAnnotations;
 
 namespace Hedwig.Models
 {
   public class FundingSpace
   {
     public int Id { get; set; }
```
Comment (cmgiven) [inline src/Hedwig/Models/FundingSpace.cs]: Should this be required as well?

File: src/Hedwig/Repositories/ChildRepository.cs [hunk 5]
```diff
@@ -88,7 +84,7 @@ public async Task<Child> GetChildByIdAsync_OLD(Guid id, DateTime? asOf = null)
 		public async Task<ILookup<int, Child>> GetChildrenByFamilyIdsAsync(IEnumerable<int> familyIds, DateTime? asOf = null)
 		{
 			var children = await GetBaseQuery<Child>(asOf)
-				.Where(c => c.FamilyId != null && familyIds.Contains((int) c.FamilyId))
+				.Where(c => familyIds.Contains((int) c.FamilyId))
```
Comment (cmgiven) [inline src/Hedwig/Repositories/ChildRepository.cs]: Hallelujah.

**Comment by cailyncodes** [comment]
Closed by #433

File: src/Hedwig/Models/FundingSpace.cs (hunk 4 above)
Comment (jlhogan) [reply src/Hedwig/Models/FundingSpace.cs]: yes! ah! will sneak that in somewhere

**Pull request closed without merging**